c10/util/Half.h

```diff
@@ -482,7 +482,7 @@ typename std::enable_if<is_complex_t<From>::value, bool>::type overflows(
 
 template <typename To, typename From>
 To checked_convert(From f, const char* name) {
-  if (overflows<To, From>(f)) {
+  if (!std::is_same<To, bool>::value && overflows<To, From>(f)) {
```

Comment:
Hmm, are you sure? I would have expected you to add more overloads to `overflows` instead.

Reply:
@ezyang I guess the question is what we expect to happen, i.e. do we still check overflow?  If we don't check overflow this implementation seems fine, but a comment explaining why we made that decision would be useful.

Reply:
Added a comment.

Reply:
What I meant was, I might reasonably expect that `convert<To, From>` is safe to call if and only if `overflows<To, From>` is false. But because you added the `is_same` check here, rather than in `overflows`, this is not true. So it really all depends on what the semantics of `overflows` here is supposed to be.  If converting to bool can never overflow, I would expect `overflows<bool, From>` to always be false.

To make matters worse, `overflows` is in the top-level namespace. So we should at least exert some effort to get this right.

Reply:
what breaks if you implement overflows properly?  The comment "  // Converting to bool can't overflow so we exclude this case from checking." doesn't really address what actually happens in PyTorch.